schema/ksl-schema.kdl

```diff
@@ -0,0 +1,925 @@
+                        allow-others
+                    }
+                }
+            }
+            node media-type about="""
+                MIME type
+
+                MIME type of string value. May be applied to 'deserialized' data
+                if value format is base64/base85 or some other stringly binary
+                encoding.
+                """ {
+                repeatable
+                args {
+                    min 1
+                    type string
+                }
+            }
+        }
+    }
+    // Number-specific validations
+    node number-validations id=number-validations {
+        ref shared-validations
+        children {
+            node div about="
+                Divisible by
+
+                Constrains them to be multiples of the given number(s). Only
+                used for numeric values. If multiple numbers are given, _any_
+                match will pass. In order to say something like `divisible by 3
+                AND by 4`, use multiple `div` nodes: `div 3; div 4`.
+                """ {
+                repeatable
+                args {
+                    min 1
+                    type number
+                }
+            }
+            node gt about="""
+                Greater than
+
+                Only used for numeric values. Constrains them to be greater than
+                the given number.
+                """ {
+                arg {
+                    type number
+                }
+            }
+            node gte about="""
+                Greater than or equal to
+
+                Only used for numeric values. Constrains them to be greater than
+                or equal to the given number.
+                """ {
+                arg {
+                    type number
+                }
+            }
+            node lt about="""
+                Less than
+
+                Only used for numeric values. Constrains them to be less than
+                the given number.
+                """ {
+                arg {
+                    type number
+                }
+            }
+            node lte about="""
+                Less than or equal to
+
+                Only used for numeric values. Constrains them to be less than or
+                equal to the given number
+                """ {
+                arg {
+                    type number
+                }
+            }
+        }
+    }
+    // Validations shared across all types.
+    node shared-validations id=shared-validations {
+        children {
+            node type about="The type for this value\n\nMultiple arguments signify a sum type." {
+                repeatable
+                args {
+                    min 1
+                    type string
+                    enum string boolean number integer #null
+                    distinct
+                }
+            }
+            // TODO: establish equality expectations.
+            node const about="""
```

Comment:
Maybe if we rename `enum`, then can get rid of this?

Maybe if `enum` were `has-value` (hmm, perhaps that name too can be improved...), it could be:
```kdl
has-value 2
has-value self documentation
has-value allow-others=#true u8 f64 date // Etc...
```

Overall, I think having, where possible, one right way to right things would be a good property and keep things simple!

Reply:
We can probably unify these, yeah.   I don't like `has-value`, though. Maybe literally just `value`.